File: src/markdown_drafter.py
"""
markdown_drafter.py

抽出テキストを Markdown 記法に整形するモジュール。
"""

import re


def draft_markdown(draft_md):
    """
    行番号付きDraft Markdownを、箇条書き・見出し・簡易表をMarkdown記法に整形する。

    Args:
        draft_md (str): L001: ... 形式の行番号付きDraft Markdown

    Returns:
        str: Markdown整形済みテキスト(行番号付き)
    """
    lines = draft_md.splitlines()
    md_lines = []
    for line in lines:
        # 行番号部分と本文に分割
        m = re.match(r"^(L\d{3}:\s)(.*)$", line)
        if not m:
            md_lines.append(line)
            continue
        prefix, content = m.groups()

        # 箇条書き
        if re.match(r"^[・\-•]\s+", content):
            content = "- " + content[2:].lstrip()
        # 見出し(例: "第1章", "1.", "1.", "1 " で始まる)
        elif re.match(r"^(第?\d+章|[0-9]+[\..\s])", content):
            content = "# " + content
        # 簡易表(タブ区切り or 全角スペース区切りが2つ以上)
        elif "\t" in content or re.search(r"　{2,}", content):
            # タブ→|、全角スペース2つ以上→|
            content = "| " + re.sub(r"(\t|　{2,})", " | ", content) + " |"

        md_lines.append(prefix + content)
    return "\n".join(md_lines)

File: src/markdown-drafter.ts
/**
 * 行番号付きDraft Markdownを、箇条書き・見出し・簡易表をMarkdown記法に整形する。
 */
export function draftMarkdown(draftMd: string): string {
	const lines = draftMd.split("\n");
	const mdLines: string[] = [];

	for (const line of lines) {
		const m = line.match(/^(L\d{3}:\s)(.*)$/);
		if (!m) {
			mdLines.push(line);
			continue;
		}
		const prefix = m[1];
		let content = m[2];

		// 箇条書き
		if (/^[・\-•]\s+/.test(content)) {
			content = `- ${content.substring(2).trimStart()}`;
		}
		// 見出し(例: "第1章", "1.", "1.", "1 " で始まる)
		else if (/^(第?\d+章|[0-9]+[.．\s])/.test(content)) {
			content = `# ${content}`;
		}
		// 簡易表(タブ区切り or 全角スペース区切りが2つ以上)
		else if (content.includes("\t") || /　{2,}/.test(content)) {
			content = `| ${content.replace(/(\t|　{2,})/g, " | ")} |`;
		}

		mdLines.push(prefix + content);
	}

	return mdLines.join("\n");
}
